src/GraphQL/Execution/ExecutionNode.cs

```diff
@@ -217,6 +217,15 @@ public interface IParentExecutionNode
         /// Returns a list of child execution nodes.
         /// </summary>
         IEnumerable<ExecutionNode> GetChildNodes();
+
+        /// <summary>
+        /// Applies the specified delegate to child execution nodes.
+        /// </summary>
+        /// <typeparam name="TState">Type of the provided state.</typeparam>
+        /// <param name="action">Delegate to execute on every child node of this node.</param>
+        /// <param name="state">An arbitrary state passed by the caller.</param>
+        /// <param name="reverse">Specifies the direct or reverse direction of child nodes traversal.</param>
+        void ApplyToChildren<TState>(Action<ExecutionNode, TState> action, TState state, bool reverse = false);
```

Comment:
The same trick here as for nodes traversal at validation stage. `GetChildNodes()` are not used now.